segment.go

```diff
@@ -472,6 +478,47 @@ func (sb *SegmentBase) dictionary(field string) (rv *Dictionary, err error) {
 	return rv, nil
 }
 
+// Thesaurus returns the thesaurus with the specified name, or an empty thesaurus if not found.
+func (s *SegmentBase) Thesaurus(name string) (segment.Thesaurus, error) {
+	thesaurus, err := s.thesaurus(name)
+	if err == nil && thesaurus == nil {
+		return emptyThesaurus, nil
+	}
+	return thesaurus, err
+}
+
+func (sb *SegmentBase) thesaurus(name string) (rv *Thesaurus, err error) {
```

Comment:
If this is called no where else than above, drop the separate method.

Reply:
done

Reply:
The `thesaurus` method is used in `mergeAndPersistSynonymSection` as it returns the type directly, the `Thesaurus` method returns an interface and is used by Bleve/Scorch... The same 2 methods are there for Dictionary as well. 
https://github.com/blevesearch/zapx/blob/cdad5b4dec7be6980c5bb6866cc9ba6b3ed552de/segment.go#L430-L479